src/reports/client/main.cpp

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+}
+
+void change_data(std::vector<Neuron>& neurons) {
+
+    // Increment in 1 per timestep every voltage
+    for (auto& neuron : neurons) {
+        for (auto& compartment: neuron.compartments) {
+            compartment++;
+        }
+    }
+}
+
+void print_data(std::vector<Neuron>& neurons) {
+
+    for (auto& neuron : neurons) {
+        std::cout << "++NEURON GID: " << neuron.gid << std::endl;
+        std::cout << "Compartments:" << std::endl;
+        for (auto& compartment: neuron.compartments) {
+            std::cout << compartment << ", ";
+        }
+        std::cout << std::endl << std::endl;
+        std::cout << "Spikes:" << std::endl;
+        for (auto& spike: neuron.spike_timestamps) {
+            std::cout << spike << ", ";
+        }
+        std::cout << std::endl;
+    }
+}
+
+int main() {
+
+    int global_rank = 0;
+#ifdef HAVE_MPI
+    std::cout << "+++++++MPI_INIT" << std::endl;
+    MPI_Init(nullptr, nullptr);
+    MPI_Comm_rank(MPI_COMM_WORLD, &global_rank);
+#endif
+    std::vector<Neuron> compartment_neurons;
+    std::vector<Neuron> soma_neurons;
+    std::vector<Neuron> spike_neurons;
+    std::vector<int> compartment_cellids;
+    std::vector<int> soma_cellids;
+    std::vector<int> spike_cellids;
+
+    std::cout << "+++++++Generating Data" << std::endl;
+    // Each rank will get different number of cells (some even 0, so will be idle ranks)
+    compartment_cellids = generate_data(compartment_neurons, "compartment", global_rank);
+    soma_cellids = generate_data(soma_neurons, "soma", global_rank);
+    spike_cellids = generate_data(spike_neurons, "spike", global_rank);
+
+    // std::cout << "+++++++Printing Data" << std::endl;
+    // print_data(spike_neurons);
+
+    std::cout << "++++++++Initializing report/cell/compartment structure" << std::endl;
+    const char* compartment_report = "compartment_report";
+    const char* soma_report = "soma_report";
+    const char* spike_report = "spike_report";
+
+    init(compartment_report, compartment_neurons);
+    init(soma_report, soma_neurons);
+    init(spike_report, spike_neurons);
+    records_set_max_buffer_size_hint(20);
+
+    std::cout << "++++++++++Prepare datasets" << std::endl;
+    records_setup_communicator();
+    records_finish_and_share();
+
+    std::cout << "++++++++++Starting simulation!" << std::endl;
+    // Calculate number of steps of the simulation
+    int num_steps = (int)std::floor((tstop - tstart) / dt);
+    if (std::fabs(num_steps * dt + tstart - tstop) > 1e-9) {
+        num_steps++;
+    }
+    double t = 0.0;
+    for(int i=0; i<num_steps; i++) {
+        std::cout << "++++Recording data for t = " << t << std::endl;
+        records_nrec(t, compartment_cellids.size(), &compartment_cellids[0], compartment_report);
+        records_nrec(t, soma_cellids.size(), &soma_cellids[0], soma_report);
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
why is this needed? Does `records_nrec` look at the time or something?

Reply:
Its just to simulate the computational phase between timesteps

Reply:
Why does that need to be simulated?

I'm confused as to what the purpose of this application is: its name doesn't really help in that regard.  Is it a test? Is it to help debug? Is it to generate fake data?  Maybe if its scope was more clear, it would make sense why we are pausing here.

Reply:
It is a test/debug application that simulates the way coreneuron would call the reportinglib.
And coreneuron has a record phase where it passes the pointer of its data to the library and a computation phase between steps. This is just trying to emulate that, but is not actually needed.